save-frontend/src/main/kotlin/com/saveourtool/save/frontend/components/views/RegistrationView.kt

```diff
@@ -0,0 +1,258 @@
+import react.dom.html.ReactHTML.form
+import react.dom.html.ReactHTML.h1
+import react.dom.html.ReactHTML.img
+import react.dom.html.ReactHTML.input
+import react.dom.html.ReactHTML.label
+import react.dom.html.ReactHTML.main
+import react.dom.html.ReactHTML.span
+
+import kotlinx.browser.window
+import kotlinx.coroutines.launch
+import kotlinx.serialization.encodeToString
+import kotlinx.serialization.json.Json
+
+/**
+ * `Props` retrieved from router
+ */
+external interface RegistrationProps : PropsWithChildren {
+    /**
+     * Currently logged in user or null
+     */
+    var userInfo: UserInfo?
+}
+
+/**
+ * [State] of registration view component
+ */
+external interface RegistrationViewState : State {
+    /**
+     * Conflict error message
+     */
+    var conflictErrorMessage: String?
+
+    /**
+     * Flag to handle uploading a file
+     */
+    var isUploading: Boolean
+
+    /**
+     * Image to owner avatar
+     */
+    var image: ImageInfo?
+
+    /**
+     * Validation of input fields
+     */
+    var isValidUserName: Boolean?
+
+    /**
+     * Map for input fields
+     */
+    var fieldsMap: MutableMap<InputTypes, String>
+}
+
+/**
+ * A Component for registration view
+ */
+@JsExport
+@OptIn(ExperimentalJsExport::class)
+class RegistrationView : AbstractView<RegistrationProps, RegistrationViewState>() {
+    init {
+        state.isValidUserName = true
+        state.isUploading = false
+        state.fieldsMap = mutableMapOf()
+    }
+
+    override fun componentDidMount() {
+        super.componentDidMount()
+        setState {
+            props.userInfo?.name?.let { fieldsMap[InputTypes.USER_NAME] = it }
+        }
+    }
+
+    override fun componentDidUpdate(prevProps: RegistrationProps, prevState: RegistrationViewState, snapshot: Any) {
+        if (props.userInfo != prevProps.userInfo) {
+            setState {
+                props.userInfo?.name?.let { fieldsMap[InputTypes.USER_NAME] = it }
+            }
+        }
+    }
+    private fun changeFields(
+        fieldName: InputTypes,
+        target: ChangeEvent<HTMLInputElement>,
+    ) {
+        val tg = target.target
+        setState {
+            fieldsMap[fieldName] = tg.value
+        }
+    }
+
+    private fun saveUser() {
+        val newUserInfo = props.userInfo?.copy(
+            name = state.fieldsMap[InputTypes.USER_NAME]?.trim() ?: props.userInfo!!.name,
+            oldName = props.userInfo!!.name,
+            isActive = true,
+        )
+
+        scope.launch {
+            val response = post(
+                "$apiUrl/users/save",
+                jsonHeaders,
+                Json.encodeToString(newUserInfo),
+                loadingHandler = ::classLoadingHandler,
+                responseHandler = ::classComponentResponseHandlerWithValidation,
+            )
+            if (response.ok) {
+                window.location.href = "#/${FrontendRoutes.PROJECTS.path}"
+                window.location.reload()
```

Comment:
why do you need to reload?

Reply:
Need to update the states in App.kt otherwise the redirect will not let us leave this view